addons/fleet/models/fleet_vehicle.py

```diff
@@ -207,13 +207,22 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
-        res = super(FleetVehicle, self).write(vals)
         if 'driver_id' in vals and vals['driver_id']:
+            self._close_driver_history()
             self.create_driver_history(vals['driver_id'])
+        res = super(FleetVehicle, self).write(vals)
+
         if 'active' in vals and not vals['active']:
             self.mapped('log_contracts').write({'active': False})
         return res
 
+    def _close_driver_history(self):
+        self.env['fleet.vehicle.assignation.log'].search([
```

Comment:
This is not correct, as we will mix all the logs from all the vehicles, as soon as they have the same drivers.

Reply:
My bad. I'm probably confused.